packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -366,7 +379,19 @@ class RenderEditable extends RenderBox {
 
   @override
   double computeMaxIntrinsicHeight(double width) {
-    return _preferredLineHeight * maxLines;
+    if (maxLines != null)
+      return _preferredLineHeight * maxLines;
+    if (width == double.INFINITY) {
+      final String text = _textPainter.text.toPlainText();
+      int lines = 1;
+      for (int index = 0; index < text.length; index += 1) {
+        if (text.codeUnitAt(index) == 0x0A) // count explicit line breaks
```

Comment:
It's strange to have this logic at this layer.  Shouldn't this logic be in text painter or lower?